cpp/order/CubaticOrderParameter.h

```diff
@@ -76,20 +170,16 @@ class CubaticOrderParameter
     float m_cubatic_order_parameter;   //!< The value of the order parameter.
     quat<float> m_cubatic_orientation; //!< The cubatic orientation.
 
-    tensor4<float> m_gen_r4_tensor;
-    tensor4<float> m_global_tensor;
-    tensor4<float> m_cubatic_tensor;
-
-    std::shared_ptr<float> m_particle_order_parameter; //!< The per-particle value of the order parameter.
-    std::shared_ptr<float>
-        m_sp_global_tensor; //!< Shared pointer for global tensor, only used to return values to Python.
-    std::shared_ptr<float>
-        m_sp_cubatic_tensor; //!< Shared pointer for cubatic tensor, only used to return values to Python.
-    std::shared_ptr<float> m_particle_tensor;
-    std::shared_ptr<float>
-        m_sp_gen_r4_tensor; //!< Shared pointer for r4 tensor, only used to return values to Python.
+    tensor4 m_gen_r4_tensor;  //!< The sum of various products of Kronecker deltas that is stored as a member for convenient reuse.
 
+    util::ManagedArray<float> m_particle_order_parameter; //!< The per-particle value of the order parameter.
+    util::ManagedArray<float>
+        m_global_tensor; //!< Copy of global tensor used to return persistent data.
+    util::ManagedArray<float>
+        m_cubatic_tensor; //!< Shared pointer for cubatic tensor, only used to return values to Python.
```

Comment:
```suggestion
        m_cubatic_tensor; //!< Cubatic tensor
```
... or something more informative.